java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/project/ProjectAlertPresenter.java

```diff
@@ -0,0 +1,622 @@
+     */
+    private static final RequestProcessor  RESOLVE_RP = new RequestProcessor(LspBrokenReferencesImpl.class.getName());
+    
+    /**
+     * Mainly a 'watcher' thread that continues the process if the user does not react on prompts.
+     */
+    private static final RequestProcessor  RP = new RequestProcessor(LspBrokenReferencesImpl.class.getName());
+    
+    private static final int WAKEUP_DELAY = 2 * 60 * 1000;
+    
+
+    /**
+     * The project.
+     */
+    private final Project project;
+    
+    /**
+     * Project name.
+     */
+    private final String projectName;
+    
+    /**
+     * Updating model of the project's problems.
+     */
+    private final BrokenReferencesModel model;
+    
+    /**
+     * The controller.
+     */
+    private final Env controller;
+    
+    /**
+     * Problems that were already seen and presented to the user.
+     */
+    private final Set<ProjectProblemsProvider.ProjectProblem> seen = Collections.synchronizedSet(new HashSet<>());
+    
+    final Object restOption = Bundle.ProjectProblems_RestOption();
+    final Object detailOption = Bundle.ProjectProblems_DetailsOption();
+
+    /**
+     * Overall result. The future will complete after all errors were seen, all questions were answered or timed out.
+     */
+    final CompletableFuture<Boolean> completion;
+    
+    /**
+     * All errors already acknowledged by the process, but not necessarily presented
+     */
+    // @GuardedBy(this)
+    Set<BrokenReferencesModel.ProblemReference> snapshot = Collections.emptySet();
+    
+    /**
+     * Invocation attempt, i.e. after a timeout or new project alert. The process continues after user response only
+     * if another run attempt has not started yet.
+     */
+    // @GuardedBy(this)
+    int runAttempt;
+
+    /**
+     * If true, will re-display project problems that have been already presented to the client.
+     */
+    boolean includeSeen;
+
+    /**
+     * Timer task that will push the process further if the user ignores the prompts. The task is run by {@link #RP} processor.
+     */
+    // @GuardedBy(this)
+    private RequestProcessor.Task wakeUpTask;
+    
+    /**
+     * The resolution process is cancelled. The process will terminate after the first timeout or user input.
+     */
+    // @GuardedBy(this)
+    private boolean cancelled;
+    
+    /**
+     * Indicates that all questions were answered. Will eventually reset to {@code false} during the process, if the user
+     * cancels further fixes.
+     */
+    // @GuardedBy(this)
+    boolean allProcessed = true;
+    
+    /**
+     * Wait threads that only record user's acknowledge for a fatal error. No real interaction is required, so there can be more
+     * of these messages to confirm
+     */
+    // @GuardedBy(RP.this)
+    RequestProcessor fatalMsgs = new RequestProcessor(ProjectAlertPresenter.class.getName(), MAX_PRESENTED_ERRORS);
```

Comment:
`RequestProcessor` with ~ ten parallel threads!? I would hope the `CompletableStage` approach leads towards a single threaded solution, not a UI that needs pool of many threads.

Reply:
I need a thread pool (now) so multiple messages can be displayed in parallel + waited on the result: `DialogDisplayer.notify()` blocks, .notifyLater does not return a value. An option would be to introduce `DialogDisplayer.notifyCompletion` that returns a `CompletableFuture`.

Reply:
Making callback on button press is the future! I have implemented that for the [HTML UI dialogs](https://bits.netbeans.org/dev/javadoc/org-netbeans-api-htmlui/org/netbeans/api/htmlui/HTMLDialog.OnSubmit.html) during NetBeans 13 development. Providing similar solution for regular dialogs is good idea.

Reply:
CompletableFuture support is proposed in #3613

Reply:
Rewritten using `CompletableFutures` in 325c279